apm-agent-core/src/test/java/co/elastic/apm/agent/MockReporter.java

```diff
@@ -164,27 +189,73 @@ public synchronized void report(Span span) {
         if (closed) {
             return;
         }
-        verifySpanSchema(asJson(dslJsonSerializer.toJsonString(span)));
-        verifyDestinationFields(span);
 
+        try {
+            verifySpanSchema(asJson(dslJsonSerializer.toJsonString(span)));
+            verifySpanType(span);
+            verifyDestinationFields(span);
+
+            if (checkUnknownOutcomes) {
+                assertThat(span.getOutcome())
+                    .describedAs("span outcome should be either success or failure for type = %s", span.getType())
+                    .isNotEqualTo(Outcome.UNKNOWN);
+            }
+        } catch (Exception e) {
+            // in case a validation error occurs, ensure that it's properly logged for easier debugging
+            e.printStackTrace(System.err);
+            throw e;
+        }
+
+        spans.add(span);
+    }
+
+
+    private void verifySpanType(Span span) {
         String type = span.getType();
-        assertThat(type).isNotNull();
+        assertThat(type)
+            .describedAs("span type is mandatory")
+            .isNotNull();
+
+        if (checkStrictSpanType) {
+            JsonNode typeJson = getMandatoryJson(SPAN_TYPES_SPEC, type, String.format("span type '%s' is not allowed by the spec", type));
+
+            String typeComment = getOptionalComment(typeJson);
+
+            JsonNode jsonOptionalSubtype = typeJson.get("optional_subtype");
+            boolean optionalSubtype = jsonOptionalSubtype != null && jsonOptionalSubtype.isBoolean() && jsonOptionalSubtype.asBoolean();
```

Comment:
`jsonOptionalSubtype.isBoolean()` should be asserted, not used as a valid initialization for `optionalSubtype`. 
In other words, `optional_subtype` should be either `null` or a string that can be parsed into a valid `boolean`, otherwise this should probably fail.